### PR TITLE
Don't regenerate items on forme change

### PR DESCRIPTION
**Pull request by AnnikaCodes**
[13:20] Tyrant: @Annika I think we shouldn’t modify item for ssb sets who change form but don’t change item (specifically: Yuki, Jho, etc)

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export function changeSet(context: Battle, pokemon: Pokemon, newSet: SSBSet, cha
 	pokemon.maxhp = newMaxHP;
 	let item = newSet.item;
 	if (typeof item !== 'string') item = item[Math.floor(Math.random() * item.length)];
-	pokemon.setItem(item);
+	if (context.toID(item) !== pokemon.item) pokemon.setItem(item);
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]: shouldnt this be `===`? otherwise it would still regenerate jho's item

Comment (AnnikaCodes) [reply data/mods/ssb/abilities.ts]: we only want to change the item if the new item (`item`) is different from the old item (`pokemon.item`)

Comment (KrisXV) [reply data/mods/ssb/abilities.ts]: yes, but when jho consumes his item, `pokemon.item` becomes `''`, right? so it would still regenerate

Comment (AnnikaCodes) [reply data/mods/ssb/abilities.ts]: Oh. Is there a way to tell what the item was before it was consumed? (Checking `if (pokemon.item)` seems too naive.)

Comment (KrisXV) [reply data/mods/ssb/abilities.ts]: `this.lastItem`?

Comment (AnnikaCodes) [inline data/mods/ssb/abilities.ts]:
```suggestion
	if (context.toID(item) !== (pokemon.item || pokemon.lastItem)) pokemon.setItem(item);
```

Comment (AnnikaCodes) [reply data/mods/ssb/abilities.ts]: sure

**Pull request merged**